qiskit/aqua/algorithms/algorithm_result.py

```diff
@@ -45,7 +45,7 @@ def pop(self, key: object, default: Optional[object] = None) -> object:
     def popitem(self) -> Tuple[object, object]:
         raise TypeError("'popitem' invalid for this object.")
 
-    def update(self, *args, **kwargs) -> None:
+    def update(self, *args, **kwargs) -> None:  # pylint: disable=arguments-differ
```

Comment:
This seems weird, if this were needed here then we would also need it on master. Maybe there was some recent change in pylint, @manoelmarques?

Reply:
I believe if one lints locally under 3.8 then this emerges as presumably the base class UserDict signature changed. This is evidently not needed by the build but then we do not lint under 3.8 at present. I had already talked to Manoel when I saw this when the PR was first done.

Reply:
yes I had 3.8,
and I reinstalled qiskit to 3.7 yesterday to avoid these conflicts.

Reply:
Ok, I raised #1135 to look at this further and maybe add 3.8 linting to the build here.